Sources/Implementation/DefaultDecisionService.swift

```diff
@@ -456,12 +442,14 @@ class DefaultDecisionService: OPTDecisionService {
                                                                       rule: experiment,
                                                                       user: user,
                                                                       userProfileTracker: userProfileTracker,
-                                                                      opType: opType,
+                                                                      isAsync: isAsync,
                                                                       options: options)
                 reasons.merge(decisionResponse.reasons)
                 if let result = decisionResponse.result {
-                    let featureDecision = FeatureDecision(experiment: experiment, variation: result.variation, source: Constants.DecisionSource.featureTest.rawValue, cmabUUID: result.cmabUUID)
-                    return DecisionResponse(result: featureDecision, reasons: reasons)
+                    if result.cmabError || result.variation != nil {
```

Comment:
Is this correct?  should be `NOT result.cmabError`? Any case variation is not nil when cmabError is true?
```suggestion
                    if result.variation != nil {
```

Reply:
> Is this correct? should be `NOT result.cmabError`? Any case variation is not nil when cmabError is true?
> 
> Suggested change
>                     if result.cmabError || result.variation != nil {
>                     if result.variation != nil {

This is intentional: when a CMAB error occurs, we return a decision with a nil variation.

Reply:
Add comment about cmab error handling